docs/source/advanced/profiler.rst

```diff
@@ -3,10 +3,291 @@
 
 .. _profiler:
 
-Performance and Bottleneck Profiler
-===================================
-.. automodule:: pytorch_lightning.profiler
-   :noindex:
-   :exclude-members:
-        _abc_impl,
-        summarize,
+#########
+Profiling
+#########
+
+Profiling your training/testing/inference run can help you understand if there are any bottlenecks in your code. The reports can be generated with ``trainer.fit()`` ``trainer.test()``, ``trainer.validate()`` and ``trainer.predict()`` with the respective actions called.
+
+
+------------
+
+****************
+Built-in Actions
+****************
+
+PyTorch Lightning supports profiling standard actions in the training loop out of the box, including:
+
+- on_epoch_start
+- on_epoch_end
+- on_batch_start
+- model_forward
+- model_backward
+- on_after_backward
+- optimizer_step
+- on_batch_end
+- training_step_end
+- on_training_end
+- etc...
+
+------------
+
+*******************
+Supported Profilers
+*******************
+
+Lightning provides the following profilers:
+
+Simple Profiler
+===============
+
+If you only wish to profile the standard actions, you can set ``profiler="simple"``. It uses the built-in
+:class:`~pytorch_lightning.profiler.simple.SimpleProfiler`.
+
+.. code-block:: python
+
+    # by passing a string
+    trainer = Trainer(..., profiler="simple")
+
+    # or by passing an instance
+    profiler = SimpleProfiler()
+    trainer = Trainer(..., profiler=profiler)
+
+The profiler's results will be printed at the completion of a training ``trainer.fit()``. Note that there are a lot more actions that will be present in the finalized
+report. Below one is just an example containing a few:
+
+.. code-block::
+
+    FIT Profiler Report
+
+    -----------------------------------------------------------------------------------------------
+    |  Action                                          |  Mean duration (s)	|  Total time (s) |
+    -----------------------------------------------------------------------------------------------
+    |  [LightningModule]BoringModel.prepare_data       |  2.112e-06      	|  2.112e-06      |
```

Comment:
This doesn't look entirely right. We should find the percentage and the list should be ordered based on total time.

Reply:
percentage and call count comes with `extended=True`(default). I tried putting it here, but it was super-ugly and nothing was clear from the output. This one is also not the original exact output. Made some alignment to render it properly in the docs.